src/components/audio_source.rs

```diff
@@ -54,7 +54,11 @@ impl AudioSinkPlayback for AudioSource {
     }
 
     fn play(&self) {
-        self.event_instance.start().unwrap();
+        if self.event_instance.get_paused().unwrap() {
+            self.event_instance.set_paused(false).unwrap();
+        } else {
+            self.event_instance.start().unwrap();
+        }
```

Comment:
I guess the if-else is not needed. You can just always set the paused state to false and then start.

Reply:
But the AudioSinkPlayback `toggle` method should resume the audio if it's paused. `toggle` calls the AudioSinkPlayback `play` method when it's currently paused. So if you were to get rid of the if-else you would restart the audio instead of resuming it.

From the Bevy source code:
```rust
    /// Toggles the playback of this sink.
    ///
    /// Will pause if playing, and will be resumed if paused.
    fn toggle(&self) {
        if self.is_paused() {
            self.play();
        } else {
            self.pause();
        }
    }
```
It's a bit of a mismatch between play/start/stop/resume behavior of Bevy Audio and FMOD.
Also they call it `play` when their intention is to resume the audio:
```rust
    /// Resumes playback of a paused sink.
    ///
    /// No effect if not paused.
    fn play(&self);
```
IMO they should just call it `resume` then.

Reply:
Sorry, I brain farted when writing that comment. You are right.